### PR TITLE
Updates build tools and simplifies pom generation

### PR DESCRIPTION
**Pull request by iambmelt**

### CODE REVIEW

File: common/build.gradle [hunk 1]
```diff
@@ -188,69 +149,49 @@ task pmd(type: Pmd) {
...
+                groupId 'com.microsoft.identity'
+                artifactId 'common'
+                //Edit the 'version' here for VSTS RC build
+                version = project.version
+
+                pom.withXml {
+                    def dependenciesNode = asNode().appendNode('dependencies')
+
+                    //Iterate over the implementation dependencies (we don't want the test ones), adding a <dependency> node for each
+                    configurations.implementation.allDependencies.each {
```
Comment (heidijinxujia) [inline common/build.gradle]: It seems this part (which is required to publish bit in public maven) is removed in this pom xml. 
```description 'This library contains code shared between the Active Directory Authentication Library (ADAL) for Android and the Microsoft Authentication Library (MSAL) for Android. This library includes only internal classes and is NOT part of the public API'
-            url 'https://github.com/AzureAD/microsoft-authentication-library-common-for-android'
-
-            developers {
-                developer {
-                    id 'microsoft'
-                    name 'Microsoft'
-                }
-            }
-
-            licenses {
-                license {
-                    name 'MIT License'
-                }
-            }
-            inceptionYear '2017'
-
-            properties {
-                branch 'master'
-                commonVersion = project.version
-            }
-
-            scm {
-                url "https://github.com/AzureAD/microsoft-authentication-library-common-for-android/tree/master"
-            }

Comment (iambmelt) [reply common/build.gradle]: Wait - do we generate separate poms for internal vs external builds?

Comment (iambmelt) [reply common/build.gradle]: And use different gradle tasks to do it?

**Comment by iambmelt** [comment]
@heidijinxujia Fixed in a1b37417b91265cc0fbd4a1aee873ac662af855a

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.microsoft.identity</groupId>
  <artifactId>common</artifactId>
  <version>0.0.1</version>
  <packaging>aar</packaging>
  <description>This library contains code shared between the Active Directory Authentication Library (ADAL) for Android and the Microsoft Authentication Library (MSAL) for Android. This library includes only internal classes and is NOT part of the public API</description>
  <name>common</name>
  <developers>
    <developer>
      <id>microsoft</id>
      <name>Microsoft</name>
    </developer>
  </developers>
  <url>https://github.com/AzureAD/microsoft-authentication-library-common-for-android</url>
  <licenses>
    <license>
      <name>MIT License</name>
    </license>
  </licenses>
  <inceptionYear>2017</inceptionYear>
  <scm>
    <url>https://github.com/AzureAD/microsoft-authentication-library-common-for-android/tree/master</url>
  </scm>
  <properties>
    <branch>master</branch>
    <version>0.0.1</version>
  </properties>
  <dependencies>
    <dependency>
      <groupId>com.android.support</groupId>
      <artifactId>appcompat-v7</artifactId>
      <version>27.1.1</version>
    </dependency>
    <dependency>
      <groupId>com.google.code.gson</groupId>
      <artifactId>gson</artifactId>
      <version>2.8.4</version>
    </dependency>
    <dependency>
      <groupId>com.nimbusds</groupId>
      <artifactId>nimbus-jose-jwt</artifactId>
      <version>5.7</version>
    </dependency>
  </dependencies>
</project>
```

**Pull request merged**